File: nodejs/sticky-session.js
const { chromium } = require("playwright");
const crypto = require("crypto");

// Generate a random 8-character hexadecimal session ID
const session = crypto.randomBytes(4).toString("hex");

(async () => {

    const browser = await chromium.launch({

        headless: true,

        proxy: {

            server: "http://resi.enigmaproxy.net:12321",

            username: `USERNAME_session-${session}`,

            password: "PASSWORD"

        }

    });

    try {

        const page = await browser.newPage();

        console.log(`Sticky Session: ${session}`);

        await page.goto("https://httpbin.org/ip", {
            waitUntil: "networkidle"
        });

        console.log(await page.locator("body").textContent());

    } finally {

        await browser.close();

    }

})();

File: python/sticky_session.py
from playwright.sync_api import sync_playwright
import secrets

session = secrets.token_hex(4)

proxy = {
    "server": "http://resi.enigmaproxy.net:12321",
    "username": f"USERNAME_session-{session}",
    "password": "PASSWORD"
}

with sync_playwright() as p:

    browser = p.chromium.launch(
        headless=True,
        proxy=proxy
    )

    try:

        page = browser.new_page()

        print(f"Sticky Session: {session}")

        page.goto(
            "https://httpbin.org/ip",
            wait_until="networkidle"
        )

        print(page.locator("body").text_content())

    finally:

        browser.close()